lib/optimizely/event/batch_event_processor.rb

```diff
@@ -90,56 +87,38 @@ def process(user_event)
         return
       end
 
-      @mutex.synchronize do
-        begin
-          @event_queue << user_event
-          @received.signal
-        rescue Exception
-          @logger.log(Logger::WARN, 'Payload not accepted by the queue.')
-          return
-        end
+      begin
+        @event_queue.push(user_event, true)
+      rescue Exception
```

Comment:
This should not be rescuing Exception. That will prevent the ruby process from safely exiting on `exit` and other base `Exception` subclasses. (should only be rescuing `StandardError`, which is the default behavior when no exception class is provided.)